editor/store/actions.js

```diff
@@ -452,60 +452,38 @@ export function initializeMetaBoxState( metaBoxes ) {
 }
 
 /**
- * Returns an action object used to signify that a meta box finished reloading.
- *
- * @param {String} location Location of meta box: 'normal', 'side' or 'advanced'.
- *
- * @returns {Object} Action object.
- */
-export function handleMetaBoxReload( location ) {
-	return {
-		type: 'HANDLE_META_BOX_RELOAD',
-		location,
-	};
-}
-
-/**
- * Returns an action object used to signify that a meta box finished loading.
- *
- * @param {String} location Location of meta box: 'normal', 'side' or 'advanced'.
+ * Returns an action object used to request meta box update.
  *
  * @returns {Object} Action object.
  */
-export function metaBoxLoaded( location ) {
+export function requestMetaBoxUpdates() {
 	return {
-		type: 'META_BOX_LOADED',
-		location,
+		type: 'REQUEST_META_BOX_UPDATES',
 	};
 }
 
 /**
- * Returns an action object used to request meta box update.
- *
- * @param {Array} locations Locations of meta boxes: ['normal', 'side', 'advanced' ].
+ * Returns an action object used signal a successfull meta nox update.
  *
  * @returns {Object} Action object.
  */
-export function requestMetaBoxUpdates( locations ) {
+export function metaBoxUpdatesSuccess() {
 	return {
-		type: 'REQUEST_META_BOX_UPDATES',
-		locations,
+		type: 'META_BOX_UPDATES_SUCCESS',
 	};
 }
 
 /**
- * Returns an action object used to set meta box state changed.
+ * Returns an action object used set the saved meta boxes data.
+ * This is used to check if the meta boxes have been touched when leaving the editor.
  *
- * @param {String}  location   Location of meta box: 'normal', 'side' or 'advanced'.
- * @param {Boolean} hasChanged Whether the meta box has changed.
- *
- * @returns {Object} Action object.
+ * @param   {Object} dataPerLocation Meta Boxes Data per location.
+ * @returns {Object}                 Action object.
  */
-export function metaBoxStateChanged( location, hasChanged ) {
+export function metaBoxSetSavedData( dataPerLocation ) {
```

Comment:
Unrelated: I prefer when the action creators are prefixed with the verb of what's happening, i.e. "set", `setMetaBoxSavedData`